driver/tke_driver.go

```diff
@@ -560,8 +593,87 @@ func getState(info *types.ClusterInfo) (*state, error) {
 
 // Update implements driver update interface
 func (d *Driver) Update(ctx context.Context, info *types.ClusterInfo, opts *types.DriverOptions) (*types.ClusterInfo, error) {
-	logrus.Info("unimplemented")
-	return nil, fmt.Errorf("not implemented")
+	logrus.Info("Invoking update cluster")
+	state, err := getState(info)
+	if err != nil {
+		return nil, err
+	}
+
+	newState, err := getStateFromOpts(opts, false)
+	if err != nil {
+		return nil, err
+	}
+
+	nodeCount := state.GoodsNum + newState.GoodsNum
+	logrus.Debugf("Updating config. ClusterName: %s, ClusterVersion: %s, NodeCount: %v", state.ClusterName, state.ClusterVersion, nodeCount)
+
+	// init tke service client
+	svc, err := d.getTKEServiceClient(ctx, state, "POST")
+	if err != nil {
+		return nil, err
+	}
+
+	if newState.GoodsNum != 0 {
+		log.Infof(ctx, "Updating node number to %v", nodeCount)
+		req, err := d.getWrapAddClusterInstancesRequest(state, newState)
+		if err != nil {
+			return nil, err
+		}
+
+		// init the TKE client
+		resp, err := svc.AddClusterInstances(req)
+		if _, ok := err.(*tcerrors.TencentCloudSDKError); ok {
+			return nil, err
+		}
+		if err == nil {
+			logrus.Infof("Add cluster instances is called for cluster %s. Status Code %v, Response Instance IDs: %s", state.ClusterID, resp.Code, resp.Data.InstanceIDs)
+		}
+		if err := d.waitTKECluster(ctx, svc, state); err != nil {
+			return nil, err
+		}
+		state.GoodsNum = nodeCount
+	}
+
+	if newState.ClusterName != "" || newState.ClusterDesc != "" {
+		log.Infof(ctx, "Updating cluster %s attributes to name: %s, desc: %s", state.ClusterName, newState.ClusterName, newState.ClusterDesc)
+		req, err := d.getWrapModifyClusterAttributesRequest(state, newState)
+		if err != nil {
+			return nil, err
+		}
+
+		// init the TKE client
+		resp, err := svc.ModifyClusterAttributes(req)
+		if _, ok := err.(*tcerrors.TencentCloudSDKError); ok {
```

Comment:
Is it possible that we catch a nonSKDError here? Looks like we simply ignore that in this case.